packages/runtime/src/widgets/Collapsible.tsx

```diff
@@ -71,9 +76,76 @@ export type CollapsibleProps = {
 } & EnsembleWidgetProps &
   HasItemTemplate;
 
+const Panel = ({
+  scope,
+  setTemplateItems,
+  template,
+}: {
+  scope: CustomScope;
+  template: CollapsibleItem;
+  setTemplateItems: React.Dispatch<React.SetStateAction<CollapsibleItem[]>>;
+}): null => {
+  const evaluated = useEvaluate({ ...template }, { context: scope });
+  const itemAddedRef = useRef(0);
+
+  const evaluatedItem = useMemo(() => {
+    const { key, label, children } = evaluated as {
+      [key: string]: Expression<string>;
+    };
+
+    const header = isString(label)
+      ? label
+      : EnsembleRuntime.render([unwrapWidget(label)]);
+
+    const content = isString(children)
+      ? children
+      : EnsembleRuntime.render([unwrapWidget(children)]);
+
+    return {
+      key,
+      label: header as Expression<string>,
+      children: content as Expression<string>,
+    };
+  }, [evaluated]) as CollapsibleItem;
+
+  useEffect(() => {
+    itemAddedRef.current += 1;
+    if (itemAddedRef.current === 2) {
+      setTemplateItems((prev) => [...prev, { ...evaluatedItem }]);
+    }
```

Comment:
If `itemAddedRef` is being used to track render count, do not do that. Update this hook to check if the item has been previously added based on the key.